### ISSUE TITLE
Remove use of `Function` constructor

### ISSUE THREAD

**Issue by sno2** [dperini/nwsapi#51]
Hello, `deno-dom` uses this library to emulate the DOM in [Deno](https://deno.land).  However, executing JavaScript from strings can result in possible insecure security flaws within code.  As a result, secure web platforms such as [Deno Deploy](https://deno.com/deploy) have disabled string code execution, which makes `nwsapi.js` unusable in secure contexts due to its [use of the `Function` constructor for codegen in nwsapi.js](https://github.com/b-fuze/deno-dom/blob/master/src/dom/nwsapi.js#L766),   I am going to try and fix this later today after I get home but just going to create an issue in response to [another issue](https://github.com/b-fuze/deno-dom/issues/75) in the `deno-dom` repo about `deno-dom` not working on Deno Deploy.

**Comment by rokoucha** [comment]
Cloudflare Workers also does not allow eval() or Function constructor.
https://developers.cloudflare.com/workers/runtime-apis/web-standards#javascript-standards

**Comment by sno2** [comment]
Here's an update: I spent a couple of days trying to refactor the code.  However, basically 700+ lines are built with generating the code that is executed in the `Function` so it took a lot of work to even get 200 lines changed to using a lexer-like system.  It was terrible debugging for me as I am used to TypeScript so I then tried to just port the code to TypeScript but there were so many styles used in the code that were above TypeScript's checker (e.g. assign var in if condition, codegen).  Therefore, I'm just going to build my own query parser that will implement all of the functionality of `nwsapi` but with simple classes and no code generation (but still allow for functional on pre-parsed selectors).  Going pretty well, but I'm going to close this issue and I expect to finish within two weeks.
